### PR TITLE
Rfc 18 application process

### PR DESCRIPTION
**Pull request by beeshot**
@chinmay241 and I worked that out. Happy to get feedback.

### CODE REVIEW

**Review by kiview — CHANGES_REQUESTED** [review]

File: governance/0018-application-process/rfc-18.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+- Status: [PRUPOSED](/README.md#proposed)
+- Since: 2020-08-12 
+- Status Note: Draft
+- Supersedes:
+- Start Date: 2020-08-01
+- Tags: governance, onboarding, member, application, process
+
+## Summary
+
+The RFC provides the guidelines to develop objective criteria for an application process to select new members to join a consortial blockchain network.
```
[the remark this answers is not in the document]
Comment (beeshot) [reply governance/0018-application-process/rfc-18.md:13]: Yes, the formal ones the requirements that need to be fulfilled to apply. This is how to evaluate them and compare them.

File: governance/0018-application-process/rfc-18.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+## Specifications
+
+* Workflow process for the organization when applying to join consortial blockchain network.
+* Needs to be combined with #0003RFC
+* Citeria to evaluate applicants
+* Voting for new members
+* Voting algorithm including a "majority rule"
+* Interval for internal voting process
+* Nods(s) that are implemented after successful application
```
[the remark this answers is not in the document]
Comment (beeshot) [reply governance/0018-application-process/rfc-18.md:28]: Nodes, so while applying it should be clear what exactly they are applying for e.g. Validator Node vs. Node without writing permission

File: governance/0018-application-process/rfc-18.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+* Transparency degree of process e.g. publication on website
```
[the remark this answers is not in the document]
Comment (beeshot) [reply governance/0018-application-process/rfc-18.md:29]: Transparency over the process and requirements. But the other option could be relevant as well

File: governance/0018-application-process/rfc-18.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+
+## Specifications
+
+* Workflow process for the organization when applying to join consortial blockchain network.
+* Needs to be combined with #0003RFC
+* Citeria to evaluate applicants
+* Voting for new members
+* Voting algorithm including a "majority rule"
+* Interval for internal voting process
```
[the remark this answers is not in the document]
Comment (beeshot) [reply governance/0018-application-process/rfc-18.md:27]: Bloxberg is an example

File: governance/0018-application-process/rfc-18.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+* Long application process may not seem contemporary for new members
+
+## Prior art
+
+* The [Sovrin Business Policy V2 Policy](https://sovrin.org/wp-content/uploads/Sovrin-Steward-Business-Policies-V2.pdf) and [Bloxberg Whitepaper v1.1](https://bloxberg.org/wp-content/uploads/2020/02/bloxberg_whitepaper_1.1.pdf) has been used to retrieve possible answers within this document. 
+
+## Unresolved questions
+
+* Is voting a good choice to approve members?
+* What alternative voting systems could be used?
```
[the remark this answers is not in the document]
Comment (beeshot) [reply governance/0018-application-process/rfc-18.md:71]: To the the one implemented by bloxberg, basically what mechanisms should be defined to emerge towards consensus in an application process.

File: governance/0018-application-process/rfc-18.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kiview) [reply governance/0018-application-process/rfc-18.md:28]: This is a technical detail, however.  So the question is, for which kind of membership does one apply (with the set of different membership classes being up to the implementation, e.g. Steward, Associated Member, Commercial Sponsor, etc.)? 

In many networks, being a member means operating a node, but e.g. bloxberg is talking about how to cater for a membership model, that allows members to join the consortium without the capabilities to operate a validator node.

File: governance/0018-application-process/rfc-18.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (kiview) [reply governance/0018-application-process/rfc-18.md:13]: However, this RFC does not (yet) specify how to evaluate and to weight the criteria outlined in RFC-0003. 

So the question is, what information goes into the decision process (as a subprocess of the application process) and what is the outcome of the decision (and if this is part of this RFC). 

A basic model would use criteria as implemented in a RFC-0003 as a binary quality gate, leading to 1 (application approved) or 0 (application dismissed). Bloxberg has a binary value (is a member an academic research organisation), however, it is evaluated by the members in a fuzzy-logic kind of way (a person is responsible to decide) and therefore the fuzziness is seemingly countered by using a majority vote.

Where does this leave us now? Maybe this hints at the indication, that we should more clearly formulate the required existence of artifacts such as a decisions process, weight functions, etc. as part of the specification.

File: governance/0018-application-process/rfc-18.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (kiview) [reply governance/0018-application-process/rfc-18.md:27]: This is true, however, by adding this point to the specification, it becomes a requirement for all implementation (and Sovrin does not perform interval voting).

Though particularly, my comment here was about `internal voting` lacking the definition in the context of the RFC. E.g. internal voting can mean voting as part of the decision process of the application process.

File: governance/0018-application-process/rfc-18.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (kiview) [reply governance/0018-application-process/rfc-18.md:71]: But this RFC does not specify the bloxberg voting mechanism, so alternative can just be in reference to the specification section.

File: governance/0018-application-process/rfc-18.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (beeshot) [reply governance/0018-application-process/rfc-18.md:27]: This was more meant as a question if internal voting should happen or not. Hence it should be discussed, As an example there is bloxberg and us an unresolved question if such a voting process really makes sense.

File: governance/0018-application-process/rfc-18.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (beeshot) [reply governance/0018-application-process/rfc-18.md:28]: Yes, this it what this implies. Can be also formulated differently. Versions of Memberships and requirements for operating node?

File: governance/0018-application-process/rfc-18.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (beeshot) [reply governance/0018-application-process/rfc-18.md:13]: how about we make it another RFC, if internal voting is considered, where weight functions, artifacts etc. described in more details. Other we add to this document.

[the remark this answers is not in the document]
Comment (kiview) [reply governance/0018-application-process/rfc-18.md]: Meaning this RFC does not handle the evaluation of the criteria as part of the applications process as implemented by an RFC-0003 implementation, correct? Actually, this RFC can have a point which is exactly this `Evaluation of applicants eligibility criteria as implemented by RFC-0003 implementation`. This is sufficiently abstract.

File: governance/0018-application-process/rfc-18.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kiview) [reply governance/0018-application-process/rfc-18.md]: These are two different points, But having something like `Classes of Membership` sounds reasonable.

File: governance/0018-application-process/rfc-18.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (beeshot) [reply governance/0018-application-process/rfc-18.md]: It is the thrid point in the spezifications: * Criteria to evaluate applicants, which would include other points than 3, like contributions to public projects (this also mentioned in the example of Soverin below). Moreover, I wrote in the unresolved questions if there is more criteria that needs to be considered. I thought providing an example is enough since the criteria probably depend strongly on the industry and use case.

File: governance/0018-application-process/rfc-18.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (beeshot) [reply governance/0018-application-process/rfc-18.md]: the requirements of the node will differ though depending on the classes of membership. Will rephrase it to classes of memberships

File: governance/0018-application-process/rfc-18.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (beeshot) [reply governance/0018-application-process/rfc-18.md]: and do we need another rfc for internal voting?

**Comment by beeshot** [comment]
it would be cool if we can finish this soon? I ask this already twice: Should we do a rfc for internal voting?

File: governance/0018-application-process/README.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+Bloxberg defines new member application workflow in its [Bloxberg Whitepaper v1.1](https://bloxberg.org/wp-content/uploads/2020/02/bloxberg_whitepaper_1.1.pdf) as follows:
+> * Voting for new bloxberg consortium members. Workflow:
+>   ** Applicant fills out a form on the official (bloxberg.org) website
+>   ** The Iron Throne adds the applicant to the voting dApp
+>   ** Once every 4 weeks members vote for 4 weeks on new applicants
+>   ** The voting is executed on-chain after an off-chain discussion
+>   ** The voting is executed according to the bloxberg decision algorithm
+>   ** The bloxberg consortium needs a quorum of >50% of all voting power, to execute a valid voting, but a minimum of 3 voting organizations.
+>   ** If > 50%of the valid votes areyes, the new member is accepted. If not,the new member is rejected and can reapply at a later time.
```
Comment (kiview) [inline governance/0018-application-process/README.md:55]:
```suggestion
>   ** If > 50%of the valid votes are yes, the new member is accepted. If not, the new member is rejected and can reapply at a later time.
```

File: governance/0018-application-process/README.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+- Start Date: 2020-08-01
+- Tags: governance, onboarding, member, application, process
+
+## Summary
+
+The RFC provides the guidelines to develop objective criteria for an application process to select new members to join a consortial blockchain network.
+ 
+## Motivation
+
+A transparent application process flow provides a clear and transparent overview to the applicant and members. It increases confidence that their application has been scrutinized well before a decision is made and that only elgibile and trustworthy organizations are able to join.
```
Comment (kiview) [inline governance/0018-application-process/README.md:17]: Please use semantic line breaks, as defined in RFC-0001.

File: governance/0018-application-process/README.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+
+## Specifications
+
+* Workflow process for the organization when applying to join consortial blockchain network.
+* Needs to be combined with #0003RFC
```
Comment (kiview) [inline governance/0018-application-process/README.md:23]: Use relative links to reference other RFCs.

File: governance/0018-application-process/README.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+The RFC provides the guidelines to develop objective criteria for an application process to select new members to join a consortial blockchain network.
+ 
+## Motivation
+
+A transparent application process flow provides a clear and transparent overview to the applicant and members. It increases confidence that their application has been scrutinized well before a decision is made and that only elgibile and trustworthy organizations are able to join.
+
+
+## Specifications
+
+* Workflow process for the organization when applying to join consortial blockchain network.
```
Comment (kiview) [inline governance/0018-application-process/README.md:22]: Process for the applying organization?

File: governance/0018-application-process/README.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+* Needs to be combined with #0003RFC
+* Criteria to evaluate applicants
+* Voting for new members
+* Voting algorithm including a "majority rule"
+* Interval for internal voting process
+* Nodes(s) that are implemented after successful application
```
Comment (kiview) [inline governance/0018-application-process/README.md:28]: Somehow all open discussions are lost on the PR? 
Nodes are a technical implementation detail, we should not consider here.

File: governance/0018-application-process/README.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+A transparent application process flow provides a clear and transparent overview to the applicant and members. It increases confidence that their application has been scrutinized well before a decision is made and that only elgibile and trustworthy organizations are able to join.
+
+
+## Specifications
+
+* Workflow process for the organization when applying to join consortial blockchain network.
+* Needs to be combined with #0003RFC
+* Criteria to evaluate applicants
+* Voting for new members
+* Voting algorithm including a "majority rule"
```
Comment (kiview) [inline governance/0018-application-process/README.md:26]: Talking about voting here is over-specification.

**Review by kiview — CHANGES_REQUESTED** [review]

File: governance/0018-application-process/rfc-18.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (kiview) [reply governance/0018-application-process/rfc-18.md]: There could be an RFC for voting, but it should one level more abstract, so decision-making mechanism. However, there might be different decision-making procedures employed for different elements of governance implementation.

So it seems to be sensible to have an abstract RFC `decision-making`, that is used as a primitive/atomic building block in other RFCs.

**Comment by kiview** [comment]
Maybe ask in the Slack or via Mail to other partners to get more reviewers involved.

File: governance/0018-application-process/README.md (hunk 10 above)
Comment (fflush) [inline governance/0018-application-process/README.md:28]: This is not quite clear to me what is actually meant in the context

File: governance/0018-application-process/README.md (hunk 11 above)
Comment (fflush) [reply governance/0018-application-process/README.md:26]: In principle I think it is not bad if the voting process is addressed but in the form that there should be voting processes.

**Review by fflush — COMMENTED** [review]
The current open points would have to be discussed again, especially the matter with the votes. It does not necessarily have to be an election, there are other possibilities. But it should still be formulated differently. Adequate criteria for the admission of new members. Otherwise I think it is good.

File: governance/0018-application-process/README.md [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+- Status: [PROPOSED](/0001-rfc-process/README.md)
+- Since: 2020-08-12 
+- Status Note: Draft
+- Supersedes:
+- Start Date: 2020-08-01
+- Tags: governance, onboarding, member, application, process
+
+## Summary
+
+The RFC provides the guidelines to develop objective criteria for an application process to select new members to join a consortial blockchain network.
```
Comment (andionita) [inline governance/0018-application-process/README.md:13]: As mentioned before, the summary is identical to the one in governance/0003-Onboarding This RFC is about the application process itself, as far as i understand.

Comment (beeshot) [reply governance/0018-application-process/README.md:13]: Can you show what is identical? I really don't see it. Plus it says criteria for the application process, which would be e.g. public contributions and are different from what we described in 0003, which are more legal requirements to join.

File: governance/0018-application-process/README.md [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+
+
+## Specifications
+
+* Workflow process for the organization when applying to join consortial blockchain network.
+* Needs to be combined with #0003RFC
+* Criteria to evaluate applicants
+* Voting for new members
+* Voting algorithm including a "majority rule"
+* Interval for internal voting process
```
Comment (andionita) [inline governance/0018-application-process/README.md:28]: I would generally replace "voting" with "decision", as it does not have to be through voting that an application is accepted or rejected.

File: governance/0018-application-process/README.md [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+- Tags: governance, onboarding, member, application, process
+
+## Summary
+
+The RFC provides the guidelines to develop objective criteria for an application process to select new members to join a consortial blockchain network.
+ 
+## Motivation
+
+A transparent application process flow provides a clear and transparent overview to the applicant and members. 
+It increases confidence that their application has been scrutinized well before a decision is made and that only elgibile and trustworthy organizations are able to join.
```
Comment (andionita) [inline governance/0018-application-process/README.md:18]: Also, a process that gives equal chances to all eligible applicants is a good motivation.

File: governance/0018-application-process/README.md [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+
+## Specifications
+
+* Workflow process for the organization when applying to join consortial blockchain network.
+* Needs to be combined with #0003RFC
+* Criteria to evaluate applicants
+* Voting for new members
+* Voting algorithm including a "majority rule"
+* Interval for internal voting process
+* Classes of Membership
```
Comment (andionita) [inline governance/0018-application-process/README.md:29]: This point is a bit vague. So if there are multiple classes of membership implies that there multiple committees deciding for each? Is this at all relevant for the application process?

Comment (beeshot) [reply governance/0018-application-process/README.md:29]: It means that there are different rights coming with different memberships (before I called it types of nodes that can be operated) than @kiview suggested classed of membership

File: governance/0018-application-process/README.md [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+>   ** The voting is executed according to the bloxberg decision algorithm
+>   ** The bloxberg consortium needs a quorum of >50% of all voting power, to execute a valid voting, but a minimum of 3 voting organizations.
+>   ** If > 50%of the valid votes are yes, the new member is accepted. If not, the new member is rejected and can reapply at a later time.
+>   ** After a positive vote the Iron Throne is instructed to add the new member as a Authority node to the bloxberg network.
+>   ** The new member has to run one Authority node, but can have n non-Authority nodes. The Authority node must be available in the network all the time.
+
+
+## Drawbacks 
+
+* On-chain voting is only pseudonym
```
Comment (andionita) [inline governance/0018-application-process/README.md:63]: Maybe rephrased as the following? In case the decision is met by members on chain, anonymity cannot be assured (pseudonimity at best).

File: governance/0018-application-process/README.md [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+* Long application process may not seem contemporary for new members
```
Comment (andionita) [inline governance/0018-application-process/README.md:64]: I see what you mean. So in case all members of the network participate in the decision making, it implies a long waiting time.

File: governance/0018-application-process/README.md [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+
+## Prior art
+
+* The [Sovrin Business Policy V2 Policy](https://sovrin.org/wp-content/uploads/Sovrin-Steward-Business-Policies-V2.pdf) and [Bloxberg Whitepaper v1.1](https://bloxberg.org/wp-content/uploads/2020/02/bloxberg_whitepaper_1.1.pdf) has been used to retrieve possible answers within this document. 
+
+## Unresolved questions
+
+* Is voting a good choice to approve members?
+* What alternative voting systems could be used?
```
Comment (andionita) [inline governance/0018-application-process/README.md:73]: Is the purpose of this RFC to describe all possible decision making schemes? I rather think not.

**Review by andionita — CHANGES_REQUESTED** [review]
Some points need to be adjusted, see comments.

File: governance/0018-application-process/README.md (hunk 15 above)
Comment (andionita) [reply governance/0018-application-process/README.md:29]: Fine. I guess it is good to have it here but somehow inside a sentence :) Maybe you rephrase your comment a bit and add it.

File: governance/0018-application-process/README.md (hunk 12 above)
Comment (andionita) [reply governance/0018-application-process/README.md:13]: You are right, they are not identical! I totally missed the "application process" part. However, I would skip the "objective criteria" part, as this applies more to the eligibility requirements. Here, it is more about developing a process. It can be mentioned later that objectivity is a property of the process.

File: governance/0018-application-process/rfc-18.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (beeshot) [reply governance/0018-application-process/rfc-18.md]: Created an issue for the decision-making RFC

File: governance/0018-application-process/README.md (hunk 18 above)
Comment (beeshot) [reply governance/0018-application-process/README.md:73]: I don't understand that comment fully :/ I created an RFC0020 called decision-making this could look at the decision-making and voting process in more detail. I think it would make this RFC more lean, understandable, and focused. Not sure if this answers your comment.

File: governance/0018-application-process/README.md (hunk 14 above)
Comment (beeshot) [reply governance/0018-application-process/README.md:18]: Also added that

File: governance/0018-application-process/README.md (hunk 18 above)
Comment (andionita) [reply governance/0018-application-process/README.md:73]: I was commenting on the fact that the unresolved questions are perhaps not relevant for this RFC, as they point to topics outside its scope, i.e. describing alternate voting schemes. But that is just my perception.

Comment (beeshot) [reply governance/0018-application-process/README.md:73]: Ahhh I see. Yes, I agree. Deleted them already, Will push the document in a few minutes.

**Review by andionita — APPROVED** [review]
Looks good to me

**Review by kiview — APPROVED** [review]
LGTM
